### PR TITLE
Hostile mob tweaks/fixes/rework

### PR DESCRIPTION
**Pull request by MadmanMartian**
There is a long list of problems with the current monster code, and I'd like to iron them all out so PVM is not as frustrating as it is currently.

### Current problems
 - Engagement with hostile mobs is bothersome, as for the most part they are critical-fail-point only, whereas the player mob is wear-over-time. This leads to most fights being incredibly one-sided.
 - The rules of engagement are different between mobs and players (Mobs can move diagonally, players can not.) so having these more consistent would make for a better experience.
 - As it stands, currently the best way to deal with a mob is to cheese it. These exploits are usually a combination of poor code and poor mapping leading to exploitable results. These should be ironed out.
 - A lot of mobs are balanced through "high numbers = harder". These should be reapproached or their usage reconsidered.

### Things done:
- In regards to destructible items. Removes the static list, adds mob flags MOBATTACK for things that can actually be attacked by the mob during escape confinement
 - reorganizes some functions, more file bloat, but less looking through an incredibly long file for a specific section.

### CODE REVIEW

**Comment by utilitas-** [comment]
Their damage should be reworked so anything decently powerful doesn't *immediately* cause your bones to break on first hit. It makes especially little sense with slimes (very squishy) glomping (high area of hit) you almost always leading to a broken leg/arm.
Furthermore, all hostile critters shouldn't just annihilate all reachable windows around them. That's stupid. They should attack only those windows in the way towards their target.

**Comment by ShiftyRail** [comment]
I don't like to fight simple mobs for a variety of reasons. I will try to list them here.

- It is very ineffective to barricade against them, since smashing a wall, a table, or even tape (post-nerf) is a free action for them, whereas all of those can make a noticeable difference in human to human combat.
- Their movement is really unpredictable ; they can sometimes lunge at you at Sanic speed, using diagonal movement, and at other times just sit still banging a wall or otherwise looking at you without doing anything. Spiders are well-known for this, but I've noticed this with zombies too; 
- Any melee attack from a simple mob is extremely punishing. You can get a broken foot, or a broken hand, or even worst, from being in melee range of any simple animal. This can be very debilitating for the rest of fights.
- Bullets have very little effect on most mobs because they do not have bones to break, or blood to lose. Conversely, lasers are extremely powerful due to their hitscan nature and the fact that you can hit and run with it. This is annoying because it means there is not a lot of options to deal with a mob outbreak.

In short, after the ta:b:e nerf, the best way to deal with simple mobs is to bring overwhelming firepower (lasers, spurs, ...), whilst on hyperzine to avoid contact with them.
Simple mobs are rarely a death sentence (when they do kill me, it's when I did something dumb, in which case I'm fine with dying), they're just annoying to fight and the removal of tape has made combat against them even more 1-dimensional.

**Comment by Caboozles** [comment]
Simple mobs more just lack the self preservation to be true threats. If they can’t get to you they’ll just stack up against whatever’s in the way, which is where the true power of lasers shines.

For example. The Syndicate Base. Very powerful hardsuited enemies that aren’t affected by the pressure or oxygen memes that are the bread and butter of vault hunters. But a few enemies DO die, and they invariably have a laser or two. 
Smash the plasmaglass windows lining the front and set up your own. Now, as long as you have enough ammo nothing can kill you and you can kill them. They can’t process shooting something behind 2 inches of glass and you can proceed to clear the entire base with a fucking window, and don’t attempt to run if unable to approach or if their oxygen supply has been cut.
With some metal alone you can wipe the “hardest” vault.

**Comment by Kurfursten** [comment]
You can reopen this if you want to work on it
